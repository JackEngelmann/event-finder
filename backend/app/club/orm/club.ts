import {
    Entity,
    Column,
    PrimaryGeneratedColumn,
    EntityRepository,
    Repository,
} from 'typeorm'

@Entity('club')
export class ClubDataModel {
    @Column({ type: 'text', nullable: true })
    address?: string | null

    @Column({ type: 'text', nullable: true })
    contact?: string | null

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    description?: string | null

    @Column({ type: 'text', nullable: true })
    email?: string | null

    @PrimaryGeneratedColumn()
    id!: number

    // LEGACY
    @Column({ type: 'text', nullable: true })
    link?: string | null

    @Column()
    name!: string

    @Column({ type: 'text', nullable: true })
    region?: string | null

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    specials?: string | null
}

@EntityRepository(ClubDataModel)
export class ClubRepository extends Repository<ClubDataModel> {
    async createAndSave(input: {
        address?: string
        contact?: string
        description?: string
        email?: string
        name: string
        region?: string
        specials?: string
    }) {
        const club = this.create(input)
        return await this.save(club)
    }
}
