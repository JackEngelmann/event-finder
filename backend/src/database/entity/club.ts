import { Entity, Column, PrimaryGeneratedColumn, Connection } from 'typeorm'
import { Logger } from '../../logger'

const logger = new Logger()

@Entity('club')
export class ClubDataModel {
    @Column({ type: 'text', nullable: true })
    address?: string | null

    @Column({ type: 'text', nullable: true })
    contact?: string | null

    @Column({ type: 'text', nullable: true })
    description?: string | null

    @Column({ type: 'text', nullable: true })
    email?: string | null

    @PrimaryGeneratedColumn()
    id!: number

    @Column({ type: 'text', nullable: true })
    imageUrl?: string | null

    @Column({ type: 'text', nullable: true })
    link?: string | null

    @Column()
    name!: string 

    @Column({ type: 'text', nullable: true })
    region?: string | null

    @Column({ type: 'text', nullable: true })
    specials?: string | null
}

export class ClubModel {
    private connection: Connection

    constructor(db: Connection) {
        this.connection = db
    }

    async createClub(input: {
        address?: string
        contact?: string
        description?: string
        email?: string
        imageUrl?: string
        link?: string
        name: string
        region?: string
        specials?: string
    }) {
        const club = new ClubDataModel()
        Object.assign(club, input)
        await this.connection.manager.save(club)
        return club.id
    }

    async updateClub(input: {
        address?: string
        contact?: string
        description?: string
        email?: string
        id: number
        imageUrl?: string
        link?: string
        name: string
        region?: string
        specials?: string
    }) {
        const club = await this.connection.manager.findOneOrFail<ClubDataModel>(
            'club',
            input.id
        )

        club.address = input.address || null
        club.contact = input.contact || null
        club.description = input.description || null
        club.email = input.email || null
        club.imageUrl = input.imageUrl || null
        club.link = input.link || null
        club.name = input.name
        club.region = input.region || null
        club.specials = input.specials || null

        await this.connection.manager.save(club)
    }

    async getClub(id: number) {
        const club = await this.connection.manager.findOne<ClubDataModel>('club', id)
        return club
    }

    async getClubs() {
        logger.info('club model: getClubs')
        const clubs = await this.connection.manager.find<ClubDataModel>('club')
        logger.info(JSON.stringify(clubs))
        return clubs
    }

    async deleteClub(id: number) {
        const club = await this.connection.manager.findOne<ClubDataModel>('club', id)
        await this.connection.manager.remove(club)
    }
}
