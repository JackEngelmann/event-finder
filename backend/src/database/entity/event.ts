import { Connection, Entity, Column, PrimaryGeneratedColumn } from 'typeorm'

@Entity('event')
export class EventDataModel {
    @Column({ type: 'decimal', nullable: true })
    admissionFee?: number | null

    @Column({ type: 'decimal', nullable: true })
    admissionFeeWithDiscount?: number | null

    @Column({ type: 'int', nullable: true })
    amountOfFloors?: number | null

    @Column()
    clubId!: number

    @Column()
    date!: string

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    description?: string | null

    @PrimaryGeneratedColumn()
    id!: number

    @Column({ type: 'text', nullable: true })
    imageUrl?: string | null

    @Column({ type: 'text', nullable: true })
    link?: string | null

    @Column({ type: 'int', nullable: true })
    minimumAge?: number | null

    @Column()
    name!: string

    @Column({ type: 'int', nullable: true })
    priceCategory?: number | null

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    special?: string | null
}

export class EventModel {
    private connection: Connection

    constructor(connection: Connection) {
        this.connection = connection
    }

    async createEvent(input: {
        admissionFee?: number
        admissionFeeWithDiscount?: number
        amountOfFloors?: number
        clubId: number
        date: string
        description?: String
        genreIds?: number[]
        imageUrl?: string
        link?: string
        minimumAge?: number
        name: string
        priceCategory?: number
        special?: string
    }) {
        const event = new EventDataModel()
        Object.assign(event, input)
        await this.connection.manager.save(event)

        return event.id
    }

    async deleteEvent(id: number) {
        const event = await this.connection.manager.findOneOrFail(
            EventDataModel,
            id
        )
        await this.connection.manager.remove(event)
    }

    async getEvent(id: number) {
        return await this.connection.manager.findOne(EventDataModel, id)
    }

    async getEvents() {
        return await this.connection.manager.find(EventDataModel)
    }

    async getEventsFromClub(clubId: number) {
        return await this.connection.manager.find(EventDataModel, { clubId })
    }

    async updateEvent(input: {
        admissionFee?: number
        admissionFeeWithDiscount?: number
        amountOfFloors?: number
        clubId: number
        date: string
        description?: string
        genreIds?: number[]
        id: number
        imageUrl?: string
        link?: string
        minimumAge?: number
        name: string
        priceCategory?: number
        special?: string
    }) {
        const event = await this.connection.manager.findOne(
            EventDataModel,
            input.id
        )
        if (!event) return undefined

        event.admissionFee = input.admissionFee || null
        event.admissionFeeWithDiscount = input.admissionFeeWithDiscount || null
        event.amountOfFloors = input.amountOfFloors || null
        event.clubId = input.clubId
        event.date = input.date
        event.description = input.description || null
        event.imageUrl = input.imageUrl || null
        event.link = input.link || null
        event.minimumAge = input.minimumAge || null
        event.name = input.name
        event.priceCategory = input.priceCategory || null
        event.special = input.special || null

        await this.connection.manager.save(event)
    }
}
