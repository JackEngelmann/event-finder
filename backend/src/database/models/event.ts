import { Database } from '../database'

export class EventDataModel {
    admissionFee?: number
    admissionFeeWithDiscount?: number
    amountOfFloors?: number
    clubId: string
    date: string
    description?: string
    id: number
    imageUrl?: string
    link?: string
    minimumAge?: number
    name: string
    priceCategory?: 1 | 2 | 3
    special?: string
    constructor(row: any) {
        this.admissionFee = row.admissionfee
        this.admissionFeeWithDiscount = row.admissionfeewithdiscount
        this.amountOfFloors = row.amountoffloors
        this.clubId = row.clubid
        this.date = row.date
        this.description = row.description
        this.id = row.id
        this.imageUrl = row.imageurl
        this.link = row.link
        this.minimumAge = row.minimumage
        this.name = row.name
        this.priceCategory = row.pricecategory
        this.special = row.special
    }
}

export class EventModel {
    private db: Database

    constructor(db: Database) {
        this.db = db
    }

    async createEvent(input: {
        admissionFee?: number
        admissionFeeWithDiscount?: number
        amountOfFloors?: number
        clubId: number
        date: string
        description?: String
        genreIds?: number[]
        imageUrl?: string
        link?: string
        minimumAge?: number
        name: string
        priceCategory?: number
        special?: string
    }) {
        return this.db
            .get(
                `
                INSERT INTO event (
                    admissionFee,
                    admissionFeeWithDiscount,
                    amountOfFloors,
                    clubId,
                    date,
                    description,
                    imageUrl,
                    link,
                    minimumAge,
                    name,
                    priceCategory,
                    special
                ) VALUES (
                    $1,
                    $2,
                    $3,
                    $4,
                    $5,
                    $6,
                    $7,
                    $8,
                    $9,
                    $10,
                    $11,
                    $12
                ) RETURNING Id
            `,
                [
                    input.admissionFee,
                    input.admissionFeeWithDiscount,
                    input.amountOfFloors,
                    input.clubId,
                    input.date,
                    input.description,
                    input.imageUrl,
                    input.link,
                    input.minimumAge,
                    input.name,
                    input.priceCategory,
                    input.special,
                ]
            )
            .then(res => res.id)
            .catch(err => {
                console.error(err)
            })
    }

    deleteEvent(id: number) {
        return this.db.run('DELETE FROM event WHERE id = $1', [id])
    }

    async getEvent(id: number) {
        const sql = 'SELECT * FROM event WHERE id = $1'
        const params = [id]
        const row = await this.db.get(sql, params)
        if (!row) return undefined
        const event = new EventDataModel(row)
        return event
    }

    async getEvents() {
        const sql = 'SELECT * FROM event'
        const rows = await this.db.all(sql, [])
        return rows.map(r => new EventDataModel(r))
    }

    async getEventsFromClub(clubId: number) {
        const sql = 'SELECT * FROM event WHERE clubId = $1'
        const rows = await this.db.all(sql, [clubId])
        return rows.map(r => new EventDataModel(r))
    }

    updateEvent(input: {
        admissionFee?: number
        admissionFeeWithDiscount?: number
        amountOfFloors?: number
        clubId: number
        date: string
        description?: String
        genreIds?: number[]
        id: number
        imageUrl?: string
        link?: string
        minimumAge?: number
        name: string
        priceCategory?: number
        special?: string
    }) {
        return this.db.run(
            `
                UPDATE event 
                SET
                    admissionFee = $1,
                    admissionFeeWithDiscount = $2,
                    amountOfFloors = $3,
                    clubId = $4,
                    date = $5,
                    description = $6,
                    imageUrl = $7,
                    link = $8,
                    minimumAge = $9,
                    name = $10,
                    priceCategory = $11,
                    special = $12
                WHERE id = $13
            `,
            [
                input.admissionFee,
                input.admissionFeeWithDiscount,
                input.amountOfFloors,
                input.clubId,
                input.date,
                input.description,
                input.imageUrl,
                input.link,
                input.minimumAge,
                input.name,
                input.priceCategory,
                input.special,
                input.id,
            ]
        )
    }
}
