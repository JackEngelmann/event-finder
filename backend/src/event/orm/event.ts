import {
    Connection,
    Entity,
    Column,
    PrimaryGeneratedColumn,
    EntityRepository,
    Repository,
} from 'typeorm'

@Entity('event')
export class EventDataModel {
    @Column({ type: 'decimal', nullable: true })
    admissionFee?: number | null

    @Column({ type: 'decimal', nullable: true })
    admissionFeeWithDiscount?: number | null

    @Column({ type: 'int', nullable: true })
    amountOfFloors?: number | null

    @Column()
    clubId!: number

    @Column()
    date!: string

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    description?: string | null

    @PrimaryGeneratedColumn()
    id!: number

    // LEGACY
    @Column({ type: 'text', nullable: true })
    link?: string | null

    @Column({ type: 'int', nullable: true })
    minimumAge?: number | null

    @Column()
    name!: string

    @Column({ type: 'int', nullable: true })
    priceCategory?: number | null

    @Column({
        type: 'text',
        nullable: true,
        charset: 'utf8mb4',
        collation: 'utf8mb4_general_ci',
    })
    special?: string | null
}

@EntityRepository(EventDataModel)
export class EventRepository extends Repository<EventDataModel> {
    async createAndSave(input: {
        admissionFee?: number
        admissionFeeWithDiscount?: number
        amountOfFloors?: number
        clubId: number
        date: string
        description?: string
        genreIds?: number[]
        minimumAge?: number
        name: string
        priceCategory?: number
        special?: string
    }) {
        const event = this.create(input)
        return await this.save(event)
    }

    async findEventsForClub(clubId: number) {
        return await this.find({ clubId })
    }
}
