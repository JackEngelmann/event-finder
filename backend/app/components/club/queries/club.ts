import { AppContext } from '../../../infrastructure/appContext'
import { ClubModel } from '../orm/club'

export function queryClub(appContext: AppContext, id: number) {
    const { db } = appContext
    const clubModel = new ClubModel(db)
    return clubModel.getClub(id)
}
