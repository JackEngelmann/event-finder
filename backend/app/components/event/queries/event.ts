import { AppContext } from '../../../infrastructure/appContext'
import { EventRepository } from '../orm/event'

export async function queryEvent(appContext: AppContext, id: number) {
    const { db } = appContext
    return await db.getCustomRepository(EventRepository).findOne({ id })
}
