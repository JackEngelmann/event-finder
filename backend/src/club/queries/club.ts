import { AppContext } from '../../infrastructure/appContext'
import { ClubRepository } from '../orm/club'

export async function queryClub(appContext: AppContext, id: number) {
    const { db } = appContext
    return await db.getCustomRepository(ClubRepository).findOne(id)
}
