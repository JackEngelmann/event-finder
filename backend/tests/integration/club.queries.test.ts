import {
    clubFragment,
    ApolloTestServer,
    createApolloTestServer,
} from '../utils'
import { ClubModel } from '../../app/components/club/orm/club'
import { EventModel } from '../../app/components/event/orm/event'
import moment = require('moment')

const DB_NAME = 'clubquerydb'

describe('club queries', () => {
    test('clubs correctly', async done => {
        // arrange
        const apolloTestServer = await createApolloTestServer({
            isAdmin: false,
            dbName: DB_NAME,
            insertTestData: true,
        })

        // act
        const result = await apolloTestServer!.client.query({
            query: `{
                clubs { ${clubFragment} }
            }`,
        })

        // assert
        expect(result.data).toBeDefined()
        expect(result).toMatchSnapshot()

        // cleanup
        await apolloTestServer.destroy()
        done()
    })
    test('existing club correctly', async done => {
        // arrange
        const apolloTestServer = await createApolloTestServer({
            isAdmin: false,
            dbName: DB_NAME,
            insertTestData: true,
        })

        // act
        const result = await apolloTestServer!.client.query({
            query: `
                query clubById($id: Int!) {
                    club(id: $id) { ${clubFragment} }
                }
            `,
            variables: {
                id: 1,
            },
        })

        // assert
        expect(result.data).toBeDefined()
        expect(result).toMatchSnapshot()

        // cleanup
        await apolloTestServer.destroy()
        done()
    })
    test('query events for a club', async done => {
        // arrange
        const apolloTestServer = await createApolloTestServer({
            isAdmin: false,
            dbName: DB_NAME,
            insertTestData: false,
        })
        const clubModel = new ClubModel(apolloTestServer.appContext.db)
        const firstClubId = await clubModel.createClub({
            name: 'club-1',
        })
        const secondClubId = await clubModel.createClub({
            name: 'club-2',
        })
        const eventModel = new EventModel(apolloTestServer.appContext.db)
        await eventModel.createEvent({
            clubId: firstClubId,
            name: 'event-club-1',
            date: moment().toISOString(),
        })
        await eventModel.createEvent({
            clubId: secondClubId,
            name: 'event-club-2',
            date: moment().toISOString(),
        })

        const result = await apolloTestServer.client.query({
            query: `
                query club($id: Int!) {
                    club(id: $id) {
                        events {
                            name
                        }
                    }
                }
            `,
            variables: {
                id: firstClubId,
            },
        })
        expect(result.errors).toBeUndefined()
        expect(result.data!.club.events.length).toBe(1)
        expect(
            result.data!.club.events.some((e: any) => e.name === 'event-club-1')
        )

        // cleanup
        await apolloTestServer.destroy()
        done()
    })
    test('query events for a club from a certain day', async done => {
        // arrange
        const apolloTestServer = await createApolloTestServer({
            isAdmin: false,
            dbName: DB_NAME,
            insertTestData: false,
        })
        const clubModel = new ClubModel(apolloTestServer.appContext.db)
        const firstClubId = await clubModel.createClub({
            name: 'club-1',
        })
        const secondClubId = await clubModel.createClub({
            name: 'club-2',
        })
        const eventModel = new EventModel(apolloTestServer.appContext.db)
        await eventModel.createEvent({
            clubId: firstClubId,
            name: 'event-club-1-1',
            date: moment()
                .subtract(1, 'day')
                .toISOString(),
        })
        await eventModel.createEvent({
            clubId: firstClubId,
            name: 'event-club-1-2',
            date: moment().toISOString(),
        })
        await eventModel.createEvent({
            clubId: secondClubId,
            name: 'event-club-2',
            date: moment().toISOString(),
        })

        const result = await apolloTestServer.client.query({
            query: `
                query club($id: Int!, $fromDay: String) {
                    club(id: $id) {
                        events(fromDay: $fromDay) {
                            name
                        }
                    }
                }
            `,
            variables: {
                id: firstClubId,
                fromDay: moment().toISOString(),
            },
        })
        expect(result.errors).toBeUndefined()
        expect(result.data!.club.events.length).toBe(1)
        expect(
            result.data!.club.events.some(
                (e: any) => e.name === 'event-club-1-2'
            )
        )

        // cleanup
        await apolloTestServer.destroy()
        done()
    })
})
