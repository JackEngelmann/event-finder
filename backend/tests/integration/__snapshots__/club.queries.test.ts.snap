// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`club queries clubs correctly 1`] = `
Object {
  "data": Object {
    "clubs": Array [
      Object {
        "address": "address",
        "contact": "contact",
        "description": "description",
        "email": "email",
        "id": 1,
        "imageUrls": Array [],
        "links": Array [],
        "name": "name",
        "region": "region",
        "specials": "specials",
      },
      Object {
        "address": "address",
        "contact": "contact",
        "description": "description",
        "email": "email",
        "id": 2,
        "imageUrls": Array [],
        "links": Array [],
        "name": "name",
        "region": "region",
        "specials": "specials",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`club queries existing club correctly 1`] = `
Object {
  "data": Object {
    "club": Object {
      "address": "address",
      "contact": "contact",
      "description": "description",
      "email": "email",
      "id": 1,
      "imageUrls": Array [],
      "links": Array [],
      "name": "name",
      "region": "region",
      "specials": "specials",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
